src/NHibernate.Test/Legacy/SQLLoaderTest.cs

```diff
@@ -140,6 +141,64 @@ public void FindBySQLProperties()
 			session.Close();
 		}
 
+		[Test]
+		public void FindBySQLDictionary()
+		{
+			using (var session = OpenSession())
+			using (var tran = session.BeginTransaction())
+			{
+				var s = new Category { Name = nextLong.ToString() };
+				nextLong++;
+				session.Save(s);
+
+				s = new Category { Name = "WannaBeFound" };
+				session.Flush();
+
+				var query =
+					session.CreateSQLQuery("select {category.*} from Category {category} where {category}.Name = :Name")
+					       .AddEntity("category", typeof(Category));
+				var parameters = new Dictionary<string, object>
+				{
+					{ nameof(s.Name), s.Name }
+				};
+				query.SetProperties(parameters);
+				var results = query.List();
+				Assert.That(results, Is.Empty);
+
+				session.Delete("from Category");
+				tran.Commit();
+			}
+		}
+
+		[Test]
+		public void FindBySQLDynamic()
+		{
+			using (var session = OpenSession())
+			using (var tran = session.BeginTransaction())
+			{
+				var s = new Category { Name = nextLong.ToString() };
+				nextLong++;
+				session.Save(s);
+
+				s = new Category { Name = "WannaBeFound" };
+				session.Flush();
+
+				var query =
+					session.CreateSQLQuery("select {category.*} from Category {category} where {category}.Name = :Name")
+					       .AddEntity("category", typeof(Category));
+				dynamic parameters = new ExpandoObject();
+				parameters.Name = s.Name;
+				// dynamic does not work on inherited interface method calls. https://stackoverflow.com/q/3071634
+				IQuery q = query;
+				q.SetProperties(parameters);
```

Comment:
A bit ugly, but there is not much we can do about it. Maybe "duplicate" the method in `ISqlQuery` in the next major release, but I do not think we should do such a thing to work around this `dynamic` short-coming.